src/alpine/3.6-ForHelix/Dockerfile

```diff
@@ -0,0 +1,89 @@
+FROM alpine:3.6
+
+RUN apk update
+
+RUN apk add --no-cache \
+        autoconf \
+        bash \
+        build-base \
+        clang \
+        clang-dev \
+        cmake \
+        coreutils \
+        curl-dev \
+        gcc \
+        gettext-dev \
+        git \
+        icu-dev \
+        krb5-dev \
+        libtool \
+        libunwind-dev \
+        libffi \
+        linux-headers \
+        llvm \
+        make \
+        openssl \
+        openssl-dev \
+        paxctl \
+        py-cffi \
+        python \
+        python-dev \
+        sudo \
+        util-linux-dev \
+        zlib-dev
+
+RUN apk -X https://dl-cdn.alpinelinux.org/alpine/edge/main add --no-cache \
+        userspace-rcu-dev \
+        lttng-ust-dev
+
+RUN apk -X https://dl-cdn.alpinelinux.org/alpine/edge/testing add --no-cache \
+        lldb-dev
+
+RUN /usr/sbin/adduser -D -G adm -s /bin/sh helixbot
+
+# Restore pip and set environment
+ADD https://bootstrap.pypa.io/get-pip.py /home/helixbot/get-pip.py
+RUN /usr/bin/python /home/helixbot/get-pip.py
+ENV pkgmgr apk
+ENV HELIX_SCRIPT_ROOT /home/helixbot/dotnetbuild/scripts
+ENV HELIX_WORK_ROOT /home/helixbot/dotnetbuild/work
+ENV HELIX_LOG_ROOT /home/helixbot/dotnetbuild/logs
+ENV HELIX_CONFIG_ROOT /home/helixbot/dotnetbuild/config
+ENV PYTHONPATH /usr/bin/python:/home/helixbot/dotnetbuild/scripts
+ENV HELIX_PYTHONPATH /usr/bin/python
+WORKDIR /home/helixbot/
+
+# Write and run script that generates starttestrunner.sh
+RUN mkdir /home/src/
+RUN echo -e "#!/bin/bash \n\
```

Comment:
I think it would be preferred to avoid the script all together and RUN these instructions directly.   If that is not plausible I think encapsulating this logic in a setup-alpine.sh script that lives next to the Dockerfile that is then added to the image via COPY would be preferred.  Writing to a script in the Docker file is not very easy to maintain and a non-standard practice.

 If you set the USER to helixbot, I wouldn't think all of the chown cmds are not needed.

Reply:
Agreed, for now, I wanted to keep everything in the docker file but having the possibility of easily extracting this logic out to a sh file and copying since we will do that once we want to do the same process for more distros. I wasn't really sure if this repo would be appropriate to host this type of dockerfiles (the ones for helix) but if so then doing what you suggest of extracting this to a file is totally doable.

Reply:
I think this is a fine place for the Helix Dockerfiles.  If you are planning to share this then yes I think encapsulating this would be preferred.

Reply:
As discussed offline, even if we used common script, it would be a fixed version of it given that we run this only once (when we build the image). Given that there is no way to get the live common script on the image, we opted to write a common script for docker images which will be re-used in the future for other containers. If we want to instead use the setup-machine-common.sh and make the required changes for it, then we can track an issue and do that later.